packages/cmf/src/cmfModule.merge.js

```diff
@@ -0,0 +1,122 @@
+import { fork } from 'redux-saga/effects';
+import { assertValueTypeOf } from './assert';
+import { noOp } from 'reassemble';
+
+function mergeObjects(obj1, obj2) {
+	if (!obj2 && obj1) {
+		return obj1;
+	}
+	if (!obj1 && obj2) {
+		return obj2;
+	}
+	return Object.keys(obj2).reduce((acc, key) => {
+		if (obj2[key] && obj1[key] !== obj2[key]) {
+			// eslint-disable-next-line no-console
+			console.warn(`override detected ${key}`);
+		}
+		return {
+			...acc,
+			[key]: obj2[key],
```

Comment:
- If `obj2[key]` is set to undefined, the previous value is still overridden and don't have the warning
- If `obj2[key]` is falsy, we don't have the warning

Reply:
I prefer throw an exception if you have set a config key to undefined.